src/openms/source/ANALYSIS/OPENSWATH/MRMIonSeries.cpp

```diff
@@ -0,0 +1,566 @@
+    {
+      if (std::fabs(ordinal->second - ProductMZ) <= mz_threshold && std::fabs(ordinal->second - ProductMZ) <= closest_delta)
+      {
+        closest_delta = std::fabs(ordinal->second - ProductMZ);
+        ion = make_pair(ordinal->first, ordinal->second);
+      }
+    }
+
+    return ion;
+  }
+
+  void MRMIonSeries::annotationToCV_(ReactionMonitoringTransition& tr)
+  {
+    OpenMS::ReactionMonitoringTransition::Product p = tr.getProduct();
+    CVTermList interpretation;
+
+    String fragment_type;
+    int fragment_nr = -1;
+    int fragment_loss = 0;
+    int fragment_gain = 0;
+
+    String annotation = tr.getMetaValue("annotation");
+
+    std::vector<String> best_annotation;
+    annotation.split("/", best_annotation);
+
+    if (best_annotation[0].find("^") != std::string::npos)
+    {
+      std::vector<String> best_annotation_charge;
+      best_annotation[0].split("^", best_annotation_charge);
+      p.setChargeState(String(best_annotation_charge[1]).toInt());
+    }
+    else
+    {
+      p.setChargeState(1);
+    }
+
+    if (best_annotation[0].find("-") != std::string::npos)
+    {
+      std::vector<String> best_annotation_loss;
+      best_annotation[0].split("-", best_annotation_loss);
+
+      fragment_type = best_annotation_loss[0].substr(0, 1);
+      fragment_nr = best_annotation_loss[0].substr(1).toInt();
+      fragment_loss = -1 * String(best_annotation_loss[1]).toInt();
+    }
+    else if (best_annotation[0].find("+") != std::string::npos)
+    {
+      std::vector<String> best_annotation_gain;
+      best_annotation[0].split("+", best_annotation_gain);
+
+      fragment_type = best_annotation_gain[0].substr(0, 1);
+      fragment_nr = best_annotation_gain[0].substr(1).toInt();
+      fragment_gain = String(best_annotation_gain[1]).toInt();
+    }
+    else
+    {
+      fragment_type = best_annotation[0].substr(0, 1);
+      fragment_nr = best_annotation[0].substr(1).toInt();
+    }
+
+    if (fragment_nr != -1)
+    {
+      CVTerm rank;
+      rank.setCVIdentifierRef("MS");
+      rank.setAccession("MS:1000926");
+      rank.setName("product interpretation rank");
+      rank.setValue(1); // we only store the best interpretation
+      interpretation.addCVTerm(rank);
+    }
+
+    if (fragment_nr != -1)
+    {
+      CVTerm frag_nr;
+      frag_nr.setCVIdentifierRef("MS");
+      frag_nr.setAccession("MS:1000903");
+      frag_nr.setName("product ion series ordinal");
+      frag_nr.setValue(fragment_nr);
+      interpretation.addCVTerm(frag_nr);
+    }
+
+    if (fragment_loss < 0)
+    {
+      CVTerm frag_loss;
+      frag_loss.setCVIdentifierRef("MS");
+      frag_loss.setAccession("MS:1001524");
+      frag_loss.setName("fragment neutral loss");
+      frag_loss.setValue(fragment_loss);
+      interpretation.addCVTerm(frag_loss);
+    }
+
+    // figure out which fragment it is
+    if (fragment_type == "x")
```

Comment:
maybe this could at some point be moved to a fxn 

`CVTerm getCVTermFromFragment(String fragment_type)`